otlp/client.go

```diff
@@ -0,0 +1,91 @@
+package otlp
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	colmetricpb "go.opentelemetry.io/proto/otlp/collector/metrics/v1"
+	"google.golang.org/protobuf/proto"
+)
+
+type Client interface {
+	Handle(context.Context, *colmetricpb.ExportMetricsServiceRequest) error
+}
+
+// HTTPClient implements the Client interface and is used to export metrics to
+// an OpenTelemetry Collector through the HTTP interface.
+//
+// The current implementation is a fire and forget approach where we do not retry
+// or buffer any failed-to-flush data on the client.
+type HTTPClient struct {
+	client   *http.Client
+	endpoint string
+}
+
+func NewHTTPClient(endpoint string) *HTTPClient {
+	return &HTTPClient{
+		//TODO: add sane default timeout configuration.
+		client:   http.DefaultClient,
+		endpoint: endpoint,
+	}
+}
+
+func (c *HTTPClient) Handle(ctx context.Context, request *colmetricpb.ExportMetricsServiceRequest) error {
+	println(request)
+	rawReq, err := proto.Marshal(request)
+	if err != nil {
+		return fmt.Errorf("failed to marshal request: %s", err)
+	}
+
+	httpReq, err := newRequest(c.endpoint, rawReq)
+	if err != nil {
+		return fmt.Errorf("failed to create HTTP request: %s", err)
+	}
+
+	return c.do(httpReq)
```

Comment:
Shouldn't we pass through the ctx.Context here? or do httpReq = httpReq.WithContext(ctx) ? 

Otherwise what's the point of passing it through?

Reply:
yep forgot to pass it through :) thanks